core/metacling/src/TClingCallFunc.cxx

```diff
@@ -1177,6 +1208,15 @@ tcling_callfunc_ctor_Wrapper_t TClingCallFunc::make_ctor_wrapper(const TClingCla
       buf << '_' << gWrapperSerial++;
       wrapper_name = buf.str();
    }
+
+   string constr_arg;
+   if (kind == ROOT::TMetaUtils::EIOCtorCategory::kIOPtrType)
+      constr_arg = "((TRootIOCtor*)nullptr)";
+   else if (kind == ROOT::TMetaUtils::EIOCtorCategory::kIORefType)
+      constr_arg = "(*((TRootIOCtor*)arena))";
+   else if (kind == ROOT::TMetaUtils::EIOCtorCategory::kIOVoidType)
+      constr_arg = "(*((__void__*)arena))";
+
```

Comment:
**_Eventually_** this code would need to be extended further to support user customizing the tag.  (see 
```
root/io/alloc/linkdef.h:#pragma link C++ ioctortype OTHER_ROOT_IO;
root/io/alloc/linkdef.h:#pragma link C++ ioctortype G4_ROOT_IO;
```
that would requires the information to be recorded by rootcling (in addition to 'just' using it as rootcling does now)

Reply:
I would say - it can be done by next iterations.

Reply:
indeed :)

Reply:
After my last changes this can be implemented relatively easy.
One can generalize `ECtorCategory` and let return more information, which can include class name or reference kind.